spec/mongoid/attributes_spec.rb

```diff
@@ -2707,4 +2707,21 @@
       catalog.set_field.should == Set.new([ 1, 2 ])
     end
   end
+
+  context 'when en embedded field has a capitalized store_as name' do
+    let(:person) { Person.new(Purse: { brand: 'Gucci' }) }
+
+    it 'sets the value' do
+      expect(person.purse.brand).to eq('Gucci')
+    end
+
+    it 'saves successfully' do
+      expect(person.save!).to eq(true)
+    end
+
+    it 'persists the value' do
+      person.save!
+      expect(person.reload.purse.brand).to eq('Gucci')
+    end
```

Comment:
Should add one more test case that the raw data in MongoDB actually has capital "P" Purse. Theoretically the code could be storing with lowercase "P" in the DB and all the tests above would still pass.